src/N3Parser.js

```diff
@@ -547,8 +559,30 @@ export default class N3Parser {
 
   // Completes a literal in subject position
   _completeSubjectLiteral(token) {
-    this._subject = this._completeLiteral(token).literal;
-    return this._readPredicateOrNamedGraph;
+    const completed = this._completeLiteral(token);
+
+    this._subject = completed.literal;
+    // If the token was consumed, continue with the rest of the input
+    if (completed.token === null)
+      return this._readPredicateOrNamedGraph;
+    // Otherwise, consume the token now
+    else {
+      return this._readPredicateOrNamedGraph(completed.token);
+    }
```

Comment:
```suggestion
    // If the token was consumed, continue with the rest of the input
    if (completed.token === null)
      return this._readPredicateOrNamedGraph;
    // Otherwise, consume the token now
    else
      return this._readPredicateOrNamedGraph(completed.token);
```

or

```suggestion
    return completed.token === null ?
      this._readPredicateOrNamedGraph :
      this._readPredicateOrNamedGraph(completed.token);
```

and the same in `_completePredicateLiteral`